app/controllers/competitions_controller.rb

```diff
@@ -272,21 +276,34 @@ def payment_integration_setup
 
   def connect_payment_integration
     competition = competition_from_params
-    payment_integration = params.require(:payment_integration)
+    payment_integration = params.require(:payment_integration).to_sym
 
     raise ActionController::RoutingError.new('Not Found') unless current_user&.can_manage_competition?(competition)
 
-    if payment_integration == 'paypal' && PaypalInterface.paypal_disabled?
+    if payment_integration == :paypal && PaypalInterface.paypal_disabled?
       flash[:error] = 'PayPal is not yet available in production environments'
       return redirect_to competitions_payment_setup_path(competition)
     end
 
-    connector = CompetitionPaymentIntegration::AVAILABLE_INTEGRATIONS[payment_integration.to_sym].safe_constantize
-    account_reference = connector&.connect_account(params)
+    if CompetitionPaymentIntegration::AVAILABLE_INTEGRATIONS[payment_integration].nil?
+      render plain: "Payment Integration #{payment_integration} not Found", status: :not_found and return
+    end
 
-    raise ActionController::RoutingError.new("Payment Integration #{payment_integration} not Found") if account_reference.blank?
+    connector = CompetitionPaymentIntegration::AVAILABLE_INTEGRATIONS[payment_integration].safe_constantize
+    integration_reference = connector&.connect_integration(params)
 
-    competition.competition_payment_integrations.new(connected_account: account_reference)
+    raise ActionController::RoutingError.new("No integration reference submitted") if integration_reference.blank?
```

Comment:
Wording on the old message was incorrect - it wasn't checking for the payment integration type, it was checking for the presence of the account reference